llvm/lib/CodeGen/AsmPrinter/AsmPrinter.cpp

```diff
@@ -3473,7 +3473,7 @@ void AsmPrinter::emitXXStructorList(const DataLayout &DL, const Constant *List,
   if (!TM.Options.UseInitArray)
     std::reverse(Structors.begin(), Structors.end());
 
-  const Align Align = DL.getPointerPrefAlignment();
+  const Align Align = DL.getPointerPrefAlignment(DL.getProgramAddressSpace());
```

Comment:
I think the only target with non-zero program address space is AVR, but not sure if it's possible to create a minimal test case there.

Reply:
I honestly don't know enough about AVR to tell.